conn.go

```diff
@@ -89,10 +89,14 @@ type SslOptions struct {
 	CertPath string
 	KeyPath  string
 	CaPath   string //optional depending on server config
-	// If you want to verify the hostname and server cert (like a wildcard for cass cluster) then you should turn this on
-	// This option is basically the inverse of InSecureSkipVerify
+
+	// If you want to verify the hostname and server cert (like a wildcard
+	// for a Cassandra cluster) then you should turn this on. This option
+	// is basically the inverse of InSecureSkipVerify. If you enable this
+	// option, make sure you set the HostServerName property too
 	// See InSecureSkipVerify in http://golang.org/pkg/crypto/tls/ for more info
 	EnableHostVerification bool
+	HostServerName         string
```

Comment:
When would a user want to set the HostServerName? Setting it here would imply that all the Cassandra nodes in the cluster have the same SNI servername.

There is also https://github.com/gocql/gocql/pull/1487 - a simple HostServerName looks insufficient for that use case, so I'd rather not introduce a new configuration option for this fix. We should export some interface in configuration that would work for that use case as well. However, that would be for a separate pull request.